wsd/RequestVettingStation.cpp

```diff
@@ -333,19 +356,21 @@ void RequestVettingStation::createClientSession(const std::string& docKey, const
     LOG_DBG("ClientSession [" << clientSession->getName() << "] for [" << docKey
                               << "] acquired for [" << url << ']');
 
-    Poco::JSON::Object::Ptr wopiInfo;
+    std::shared_ptr<std::unique_ptr<WopiStorage::WOPIFileInfo>> wopiFileInfo;
 #if !MOBILEAPP
     assert((!_checkFileInfo || _checkFileInfo->wopiInfo()) &&
            "Must have WopiInfo when CheckFileInfo exists");
-    wopiInfo = _checkFileInfo ? _checkFileInfo->wopiInfo() : nullptr;
+    // unique_ptr is not copyable, so cannot be captured in a std::function-wrapped lambda.
+    wopiFileInfo = std::make_shared<std::unique_ptr<WopiStorage::WOPIFileInfo>>(
```

Comment:
Surely there is no benefit at all to a shared_ptr around a unique_ptr - can we not just use a shared_ptr without that ? =)

Reply:
We can, but the current design removes entries from the wrapped JSON. So, many places need to be changed from unique to shared, but we would still have the questionable design of modifying the JSON from a shared object.

Again, refrained from making too many changes. This PR is already complex and large as it is. Wanted to get it working rather than beautifying.

Agreed it needs a second pass.